### ISSUE TITLE
Update Pydantic to 2.x

### ISSUE THREAD

**Issue by benjamin-kirkbride** [PurpleMyst/sansio-lsp-client#55]
I need Pydantic 2.x for a thing I'm working on but "sansio-lsp-client 0.10.0 requires pydantic<2.0.0,>=1.7.3"

**Comment by PurpleMyst** [comment]
Our usage of `pydantic` is rather easy to migrate, at it seems we'd only need to update our usage of `Config` and `parse_obj_as`. I'm curious what @Akuli thinks of the matter, as one option is just detecting at runtime if `pydantic>=2.0.0` is installed and, if so, importing everything from the `pydantic.v1` module. That way we'd practically be version agnostic, apart from some more subtle changes we'd have to review.

**Comment by Akuli** [comment]
I don't think we need to keep v1 support, unless someone specifically asks for it, and in that case `pydantic.v1` is probably the best option.

Also, nice to see that you are still around :)

**Comment by pragone** [comment]
Hi,
Thanks for your work... I tried to use this library and encountered that it was conflicting with other requirements because it depends on pydantic v1.
So, I decided to give back to the community. Hopefully this is useful and looking forward to any feedback.
I made sure that all tests pass, that all strict-mode checks pass.
Cheers

**Comment by pragone** [comment]
Apologies... it seems that there's a few more things to fix before it's ready for review.

**Comment by pragone** [comment]
Done... it's all ready for review now
